packages/addon-dev/src/rollup-hbs-plugin.ts

```diff
@@ -30,9 +30,24 @@ export default function rollupHbsPlugin({
       }
     },
 
+    transform(code: string, id: string) {
+      let hbsFilename = id.replace(/\.\w{1,3}$/, '') + '.hbs';
+      if (hbsFilename !== id) {
+        this.addWatchFile(hbsFilename);
+      }
+      if (hbsFilter(id)) {
+        let basename = id.replace(/\.\w{1,3}$/, '');
+        this.addWatchFile(basename + '.ts');
+        this.addWatchFile(basename + '.js');
```

Comment:
I think these belong beside the spot where we actually attempted to resolve the corresponding JS files in `maybeSynthesizeJS`. Having them together may address the caching behaviors you mentioned that caused you to have to move code between load and transform phases.

Reply:
I tried to move it to the resolve hook. But then the transform for js files does not get updated. It might need to be in the transform hook for the cache to be invalidated